kfs-core/src/main/java/edu/arizona/kfs/fp/document/validation/impl/GeneralErrorCorrectionErrorCertificationValidation.java

```diff
@@ -0,0 +1,191 @@
+	
+	private ParameterService parameterService;
+	private BusinessObjectService businessObjectService;
+	private AccountingLineBase accountingLineForValidation;
+	
+	protected static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(GeneralErrorCorrectionErrorCertificationValidation.class);
+	
+	public boolean validate(AttributedDocumentEvent event) {
+		LOG.debug("validate error certification tab called");
+		GeneralErrorCorrectionDocument currDocument = (GeneralErrorCorrectionDocument)event.getDocument();
+		String defaultDaysFromParam = parameterService.getParameterValueAsString(currDocument.getClass(), KFSParameterKeyConstants.FpParameterConstants.DEFAULT_NUMBER_OF_DAYS_ERROR_CERTIFICATION_TAB_REQUIRED);
+		Integer daysFromParameter = new Integer(defaultDaysFromParam);
+		
+		boolean success = true;
+		boolean olderThanOriginal = true;
+		boolean checkSubFund = false; // indicates whether sub fund should be checked for ERROR_CERTIFICATION_DEFAULT_OVERRIDE_BY_SUB_FUND rule
+		int numBlankFields = 0;
+		List<AccountingLineBase> acctLines;
+		
+		// check original transaction dates and the system parameters involving age of transaction
+		try {
+			acctLines = currDocument.getSourceAccountingLines();
+			olderThanOriginal = !defaultNumberOfDaysCheck(currDocument, acctLines, daysFromParameter, checkSubFund);
+			LOG.debug("olderThanOriginal for source accounting lines: " + olderThanOriginal);
+			
+			// if all source accounting lines are younger than the days specified in the parameter, check target accounting lines
+			if(olderThanOriginal) {
+				acctLines = currDocument.getTargetAccountingLines();
+				checkSubFund = true;
+				olderThanOriginal = !defaultNumberOfDaysCheck(currDocument, acctLines, daysFromParameter, checkSubFund);
+				LOG.debug("olderThanOriginal for target accounting lines: " + olderThanOriginal);
+			}
+		} catch (EntryNotFoundException e) {
+			GlobalVariables.getMessageMap().putError(KFSPropertyConstants.REFERENCE_NUMBER, KFSKeyConstants.ERROR_GEC_REF_NUMBER_INVALID, e.getCriteria().get(KRADPropertyConstants.DOCUMENT_NUMBER));
+			LOG.debug("defaultNumberOfDaysCheck failed", e);
+			success = false;
+		}
+		
+		// check for blank fields in the Error Certification Tab
+		ErrorCertificationValidation errorCertificationTabValidation = new ErrorCertificationValidation();
+		numBlankFields = errorCertificationTabValidation.errorCertificationBlankFieldsCheck(currDocument);
+		LOG.debug("numBlankFields: " + numBlankFields);
+		
+		// if there is a transaction older than DEFAULT_NUMBER_OF_DAYS_ERROR_CERTIFICATION_TAB_REQUIRED and Error Certification tab is partially filled
+		if (olderThanOriginal && ((numBlankFields > 0) && (numBlankFields <= KFSConstants.ErrorCertificationConstants.NUM_ERROR_CERT_FIELDS))) {
+			GlobalVariables.getMessageMap().putErrorForSectionId("document.errorCertification", KFSKeyConstants.ERROR_ERROR_CERT_DATE_PARAM_TRIGGERED, "error.errorCert.dateParamTriggered");
```

Comment:
Magic string "error.errorCert.dateParamTriggered" can be pulled from edu.arizona.kfs.sys.KFSKeyConstants.ERROR_ERROR_CERT_DATE_PARAM_TRIGGERED.